ui/src/App.tsx

```diff
@@ -404,8 +404,10 @@ export function App() {
   ) => {
     if (actionSuccessfullyCompleted) {
       const rowsCopy = rows.slice();
+      const highestId = rows.slice(-1)[0].id;
```

Comment:
isn't it similar to
```suggestion
      const highestId = rows.pop()?.id;
```
I have added a `?` because if rows is empty I think pop returns undefined

Reply:
Oh yes, you are right!
Initially I was looking  for an alternative to the `[0]` to avoid any potential error. But accessing an index that does not exist returns undefined. so I guess what could be missing here is only the `?`